mitodl/open-discussions
# fix drawer z-index issue

#### Pre-Flight checklist


#### What are the relevant tickets?

closes #1274 

#### What's this PR do?

This fixes an issue with the z-index on the mobile nav drawer, where it was rendering underneath the `<Toolbar />` component. I also pulled the hambuger button and the MIT logo out of the two different navbars into one component, which can be shared between them, so it will be easier to change in the future.

#### How should this be manually tested?

Make sure that you can open and close the drawer at all widths, and make sure that the mobile nav drawer floats over the toolbar.
